tox.ini

```diff
@@ -6,7 +6,10 @@ whitelist_externals = source
                       bash
 install_command = pip install -U {opts} {packages}
 setenv = TOX_ENV_NAME={envname}
-passenv = TOX_* TRANSIFEX_USER TRANSIFEX_PASSWORD CI_* APPVEYOR_*
+passenv = TOX_* TRANSIFEX_USER TRANSIFEX_PASSWORD CI CI_* CIRCLECI CIRCLE* APPVEYOR*
+deps = codecov>=1.4.0
+       coverage
 commands = python -V
-           python setup.py test
+           coverage run --append setup.py test
            bash ./contrib/test_build.sh
+           codecov -e TOX_ENV_NAME --token=5ad93ecd-ffdc-4c92-b2db-a7c0ff1c1571
```

Comment:
@dimoschi shouldn't we put the codecov token in the CircleCI environment? And maybe disable codecov when running on AppVeyor?